k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSetupIncoming.java

```diff
@@ -538,55 +554,96 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     protected void onNext() {
-        try {
-            ConnectionSecurity connectionSecurity = getSelectedSecurity();
+        AuthType authType = getSelectedAuthType();
+        if (authType == AuthType.XOAUTH2) {
+            K9.oAuth2TokenStore.authorizeAPI(mAccount.getEmail(), this,
```

Comment:
But it makes it less clear that you're waiting for the API authorization before continuing (the success block takes the place of the stuff done if XOAUTH isn't being used.